### ISSUE TITLE
a11y: Fix keyboard navigation, aria-label, role and focus display for user preferences screen

### ISSUE THREAD

**Issue by nwmac** [rancher/dashboard#12791]
a11y Issues: 21

Keyboard navigation for the combo box used in these places:

- Language
- Date Format
- Time Format and so on

does not have correct keyboard navigation.

Also missing the role, aria-label etc attributes.

![image](https://github.com/user-attachments/assets/8b53878c-2da9-477d-b78e-cadd938c02a9)

**Comment by IsaSih** [comment]
Reopening the issue as some components of the preferences screen don't offer consistent keyboard navigation as seen in the screen recording:

1- The Languge dropdown doesn't support navigation both with arrow keys and tab. Also, doesn't allow selection with the spacebar
2- Login Landing Page round buttons supports arrow key selects, but it's not possible to tab between the options
3- All Display Settings dropdowns: Date Format, Time Format and Table Rows per Page are not supporting navigation both with arrow keys and tab. Also, doesn't allow selection with the spacebar

https://github.com/user-attachments/assets/0bb384a8-3d25-4671-8409-e0e6a0918211

**Comment by aalves08** [comment]
> Reopening the issue as some components of the preferences screen don't offer consistent keyboard navigation as seen in the screen recording:
> 
> 1- The Languge dropdown doesn't support navigation both with arrow keys and tab. Also, doesn't allow selection with the spacebar 2- Login Landing Page round buttons supports arrow key selects, but it's not possible to tab between the options 3- All Display Settings dropdowns: Date Format, Time Format and Table Rows per Page are not supporting navigation both with arrow keys and tab. Also, doesn't allow selection with the spacebar
> 
>  Screen.Recording.2025-03-03.at.19.33.57.mov

Hi @IsaSih . Thank you for testing this. As for the points you've mentioned:

1 - The language dropdown is what W3C refers to as a `combobox`. If you take a look at the documentation https://www.w3.org/WAI/ARIA/apg/patterns/combobox/ , specifically in the chapter `Listbox Popup Keyboard Interaction`, there's no mention of the `space` key a "selector" of an option, nor `tab` being able to cycle through different options. 

2 - For a `radio group`, according to https://www.w3.org/WAI/ARIA/apg/patterns/radio/ there's no interaction with the `tab` key, but only with the arrow keys. This got fixed quite recently with https://github.com/rancher/dashboard/issues/13427 + https://github.com/rancher/dashboard/pull/13491  because I've noticed the wrong behaviour

3 - Same comment as `1-`

FYI @rak-phillip

**Comment by IsaSih** [comment]
Thanks for the clarification @aalves08. However, I got confused because by using the up/down arrow keys, the UI highlights the option that is not in focus, as shown in the screen recording below. In my opinion, this needs to be fixed. 

https://github.com/user-attachments/assets/55e8fafc-cb41-4758-889b-a61f1a4ffdc8

**Comment by rak-phillip** [comment]
@IsaSih please see this comment from @edenhernandez-suse regarding the styling of the options

https://github.com/rancher/dashboard/issues/13389#issuecomment-2657017917

I don't think there's an issue here, the darker highlight is signalling that "English" is the currently selected language in the list.

**Comment by IsaSih** [comment]
Okay, Phill. Thanks for the clarification. I'll close the issue.
